File: user-management/create_users.py
import sys
import subprocess
import json

user_pool_id = sys.argv[1]
profile_name = sys.argv[2] if len(sys.argv) > 2 else 'default'

def json_input(user):
    return {
        'Username': user["username"] if "username" in user else user["name"],
        'UserPoolId': user_pool_id,
        'ValidationData': [],
        'UserAttributes': [
            {'Name': 'email', 'Value': user["email"]},
            {'Name': 'email_verified', 'Value': user["emailVerified"] if "emailVerified" in user else "false"},
            {'Name': 'phone_number', 'Value': user["phoneNumber"]},
            {'Name': 'phone_number_verified', 'Value': user["phoneNumberVerified"] if "phoneNumberVerified" in user else "false"},
            {'Name': 'custom:userUUID', 'Value': user["uuid"]}
        ],
        'ForceAliasCreation': True,
        'TemporaryPassword': user["password"],
        'DesiredDeliveryMediums': ['SMS', 'EMAIL']
    }

def create_user(user):
    user_json = json_input(user)
    print user_json
    try:
        print subprocess.check_output(
            ["aws", "--profile", profile_name, "cognito-idp", "admin-create-user", "--cli-input-json", json.dumps(user_json)])
    except Exception as e:
        print e, '\n'

users = json.load(open('./users.json'))
map(create_user, users)

File: user-management/createUsers.js
//NOT USED
let users = require("./users");
const { spawnSync } = require("child_process");

let cognitoUsers = users.map(function (user) {
    return {
        'Username': user["name"],
        'UserPoolId': process.argv[2],
        'ValidationData': [],
        'UserAttributes': [
            {'Name': 'email', 'Value': user["email"]},
            {'Name': 'phone_number', 'Value': user["phoneNumber"]},
            {'Name': 'custom:organisationName', 'Value': user["orgName"]},
            {'Name': 'custom:isAdmin', 'Value': `${user["admin"]}`},
            {'Name': 'custom:isOrganisationAdmin', 'Value': `${user["orgAdmin"]}`},
            {'Name': 'custom:userUUID', 'Value': user["uuid"]}
        ],
        'ForceAliasCreation': true,
        'TemporaryPassword': user["password"],
        'DesiredDeliveryMediums': ['SMS', 'EMAIL']
    };
});


cognitoUsers.forEach(function (cognitoUser) {
    let aws = spawnSync('aws', ["cognito-idp", "admin-create-user", "--cli-input-json", cognitoUser]);
    console.log(aws.output);
    // aws.stdout.on('data', (data) => {
    //     console.log(`stdout: ${data}`);
    // });
    // aws.stderr.on('data', (data) => {
    //     console.log(`stderr: ${data}`);
    // });
    // aws.on('close', (code) => {
    //     console.log(`child process exited with code ${code}`);
    // });
});

